b-real-time/b-real-time-htb.js

```diff
@@ -99,149 +108,97 @@ function BRealTimeHtb(configs) {
-         *     },
-         *     callbackId: '_23sd2ij4i1' //unique id used for pairing requests and responses
-         * }
-         */
-
-
         /* PUT CODE HERE */
 
-        var __tags = [];
+        var __emxBids = [];
+        var __emxData = {};
         for (var i = 0; i < returnParcels.length; i++) {
-            var returnParcel = returnParcels[i],
-                tag = {},
-                uuid = System.generateUniqueId();
-            tag.sizes = returnParcel.xSlotRef.sizes.map(function (obj) {
-                return {
-                    width: obj[0],
-                    height: obj[1]
-                };
-            });
-            tag.id = returnParcel.xSlotRef.placementId;
-            tag.primary_size = tag.sizes[0];
-            tag.allow_smaller_sizes = false;
-            tag.prebid = true;
-            tag.disable_psa = true;
-            tag.uuid = uuid;
-            returnParcel.uuid = uuid;
-
-            if (gdprPrivacyEnabled) {
-              if (gdprStatus.consentString !== void 0) {
-                tag.gdpr_consent = gdprStatus.consentString;
-              }
-              if (gdprStatus.applies !== void 0) {
-                tag.gpdr = gdprStatus.applies ? "1" : "0";
-              }
+            var returnParcel = returnParcels[i];
+            var emxBid = {};
+            var bidFloor = returnParcel.xSlotRef.bidfloor;
+
+            emxBid.id = returnParcels[i].xSlotName;
+            emxBid.tagid = returnParcel.xSlotRef.tagid;
+            emxBid.secure = networkProtocol;
+            if (bidFloor > 0) {
+                emxBid.bidfloor = bidFloor;
             }
-
-            __tags.push(tag);
+            emxBid.banner = {
+                format: returnParcel.xSlotRef.sizes.map(function (size) {
+                    return {
+                        w: size[0],
+                        h: size[1]
+                    };
+                }),
+                w: returnParcel.xSlotRef.sizes[0][0],
+                h: returnParcel.xSlotRef.sizes[0][1]
+            };
+
+            __emxBids.push(emxBid);
         }
 
+        __emxData = {
+            id: callbackId,
+            imp: __emxBids,
+            site: {
+                domain: pageHost,
+                page: pageUrl,
+                ref: Browser.getReferrer()
+            },
+            ext: {
+                ver: version
+            }
+        };
 
+        if (gdprPrivacyEnabled) {
+            /* eslint-disable camelcase */
+            if (gdprStatus.hasOwnProperty('consentString')) {
+                __emxData.user = {
+                    ext: {
+                        consent: gdprStatus.consentString
+                    }
+                };
+            }
+            /* eslint-enable camelcase */
+
+            if (gdprStatus.hasOwnProperty('applies')) {
+                __emxData.regs = {
+                    ext: {
+                        gdpr: gdprStatus.gdprApplies ? 1 : 0
```

Comment:
Small typo here, `.gdprApplies` should be just `.applies` as per the template documetnation:

```plaintext
The return object should look something like this:
{
    applies: true,
    consentString: "BOQ7WlgOQ7WlgABABwAAABJOACgACAAQABA"
}
```

Reply:
Thanks for catching, this has been fixed.